CameraColorPicker/app/src/main/java/fr/tvbarthel/apps/cameracolorpicker/activities/PaletteCreationBaseActivity.java

```diff
@@ -100,45 +135,21 @@ public void onClick(View v) {
     }
 
     @Override
-    public void onEditTextDialogFragmentPositiveButtonClick(int requestCode, String text) {
-        final Palette newPalette = mPaletteMakerView.make(text);
-        if (Palettes.saveColorPalette(this, newPalette)) {
-            finish();
-        }
-    }
-
-    @Override
     public void onColorItemClicked(@NonNull ColorItem colorItem, @NonNull View colorPreview) {
         mPaletteMakerView.addColor(colorItem);
         if (mPaletteMakerView.size() == 1) {
             mRemoveLastColorBtnAnimator.setFloatValues(1f);
             mRemoveLastColorBtnAnimator.start();
-        }
-    }
 
-    @Override
-    public void onEditTextDialogFragmentNegativeButtonClick(int requestCode) {
-        // Do nothing when the user cancels.
-    }
-
-    /**
-     * Hide the current {@link android.widget.Toast}.
-     */
-    private void hideToast() {
-        if (mToast != null) {
-            mToast.cancel();
-            mToast = null;
+            mFab.animate()
+                    .translationY(0)
+                    .setDuration(FAB_ANIMATION_DURATION)
+                    .setInterpolator(mDecelerateInterpolator);
         }
     }
 
     /**
-     * Show a toast text message.
-     *
-     * @param resId The resource id of the string resource to use.
+     * Create a palette.
      */
-    private void showToast(@StringRes int resId) {
-        hideToast();
-        mToast = Toast.makeText(this, resId, Toast.LENGTH_SHORT);
-        mToast.show();
-    }
+    protected abstract void createPalette();
```

Comment:
would add a small comment for maintainability : called when user validate the palette creation.